### ISSUE TITLE
Branching strategy

### ISSUE THREAD

**Issue by tdudgeon** [m2ms/fragalysis-frontend#982]
When Ruben mentioned that we could not deploy the "green" release as things from the "purple" release had gone into it, it made me realise that the branching strategy that we are using is not correct. We should not get into that situation.

Instead we should create a branch for each planned release. In this case we would create a temporary *green* branch and all feature branches (e.g. #123) relating to that release should be merged into it.

If we then started on a "purple" release before the *green* one was complete then we create a temporary *purple* branch off the *green* branch so that any subsequent changes on the *green* branch can easily be merged into *purple*.

When the *green* branch is ready for testing it gets merged into a permanent *staging* branch from where things are deployed to the staging environment.
Once testing is complete the *staging* branch gets merged into a permanent *production* branch from where things are tagged (using the release number) and deployed to the production environment. (Alternative - we use *master* instead of *production*).

When the green release is complete the *green* branch is deleted (so that the name can be re-used later).

If the purple release is started after the *green* branch is deleted then it is taken off *staging*.

This way even though commits on the purple release have been made, the *green* branch is unafected by this so can be deployed at any stage.

This policy would need to be applied to all the repos invovled (FE, BE, stack) in order to be consistent.

**Comment by tdudgeon** [comment]
Duplicate of #192
